ydb/core/kqp/opt/logical/kqp_opt_log_indexes.cpp

```diff
@@ -139,6 +145,65 @@ bool CanPushTopSort(const TCoTopBase& node, const TKikimrTableDescription& index
     return IsTableExistsKeySelector(node.KeySelectorLambda(), indexDesc, columns);
 }
 
+bool CanUseVectorIndex(const TIndexDescription& indexDesc, const TExprBase& lambdaBody, const TCoTopBase& top) {
+    Y_ASSERT(indexDesc.Type == TIndexDescription::EType::GlobalSyncVectorKMeansTree);
+    // TODO(mbkkt) We need to account top.Count(), but not clear what to if it's value is runtime?
+    auto checkMember = [&] (const TExprBase& expr) {
+        auto member = expr.Maybe<TCoMember>();
+        return member && member.Cast().Name().Value() == indexDesc.KeyColumns[0];
+    };
+    auto checkUdf = [&] (const TExprBase& expr, bool checkMembers) {
+        auto apply = expr.Maybe<TCoApply>();
+        if (!apply || apply.Cast().Args().Count() != 3) {
+            return false;
+        }
+        if (checkMembers) {
+            auto args = apply.Cast().Args();
+            if (absl::c_none_of(args, [&] (const TExprBase& expr) { return checkMember(expr); })) {
```

Comment:
absl::c_none deps indirectly comes from flat hash map? just question not a problem.

Reply:
Yes, absl algorithm header, it's reuse std::none_of